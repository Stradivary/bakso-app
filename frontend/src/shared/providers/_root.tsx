import { MantineProvider } from "@mantine/core";
import { Notifications } from "@mantine/notifications";
import { AuthProvider } from "./authProvider";

import "@mantine/core/styles.css";
import "@mantine/notifications/styles.css";
import { RouteProvider } from "./routeProvider";


export function Root() {
  return (
    <AuthProvider>
      <MantineProvider theme={{ primaryColor: "red" }}>
        <RouteProvider />
        <Notifications />
      </MantineProvider>
    </AuthProvider>
  );
}

