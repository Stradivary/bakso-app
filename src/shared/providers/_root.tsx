import "@mantine/core/styles.css";
import "@mantine/notifications/styles.css";
import "@fontsource/poppins/index.css";

import { AuthProvider } from "./authProvider";
import { RouteProvider } from "./routeProvider";
import { ThemeProvider } from "./themeProvider";
import { PropsWithChildren } from "react";

export function Contexts({ children }: PropsWithChildren) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export function Root() {
  return (
    <AuthProvider>
      <RouteProvider />
    </AuthProvider>
  );
}
