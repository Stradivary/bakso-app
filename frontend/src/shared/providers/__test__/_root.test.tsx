import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import { Root } from "../_root";

// Mock the RouteProvider component
vi.mock("../routeProvider", () => ({
  RouteProvider: () => <div>Mocked RouteProvider</div>,
}));

describe("Root component", () => {
  it("renders correctly", () => {
    const r = render(<Root />);

    expect(r).toMatchSnapshot();
  });
});
