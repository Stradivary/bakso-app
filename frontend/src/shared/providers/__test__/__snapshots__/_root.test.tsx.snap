// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Root component > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-mantine-styles="true"
      >
        :root{--mantine-primary-color-filled: var(--mantine-color-red-filled);--mantine-primary-color-filled-hover: var(--mantine-color-red-filled-hover);--mantine-primary-color-light: var(--mantine-color-red-light);--mantine-primary-color-light-hover: var(--mantine-color-red-light-hover);--mantine-primary-color-light-color: var(--mantine-color-red-light-color);--mantine-primary-color-0: var(--mantine-color-red-0);--mantine-primary-color-1: var(--mantine-color-red-1);--mantine-primary-color-2: var(--mantine-color-red-2);--mantine-primary-color-3: var(--mantine-color-red-3);--mantine-primary-color-4: var(--mantine-color-red-4);--mantine-primary-color-5: var(--mantine-color-red-5);--mantine-primary-color-6: var(--mantine-color-red-6);--mantine-primary-color-7: var(--mantine-color-red-7);--mantine-primary-color-8: var(--mantine-color-red-8);--mantine-primary-color-9: var(--mantine-color-red-9);}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-red-4);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-red-6);}
      </style>
      <style
        data-mantine-styles="classes"
      >
        @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
      </style>
      <div>
        Mocked RouteProvider
      </div>
    </div>
    <div
      data-portal="true"
    >
      <div>
        <div
          class="m_9df02822 mantine-Modal-root"
          style="--modal-y-offset: 5dvh; --mb-z-index: 201;"
        />
      </div>
    </div>
    <div
      data-portal="true"
    >
      <div>
        <div
          class="m_9df02822 mantine-Modal-root"
          style="--modal-y-offset: 5dvh; --mb-z-index: 200;"
        />
      </div>
    </div>
    <div
      data-portal="true"
    >
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="top-center"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div />
      </div>
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="top-left"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div />
      </div>
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="top-right"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div />
      </div>
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="bottom-right"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div />
      </div>
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="bottom-left"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div />
      </div>
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="bottom-center"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <style
      data-mantine-styles="true"
    >
      :root{--mantine-primary-color-filled: var(--mantine-color-red-filled);--mantine-primary-color-filled-hover: var(--mantine-color-red-filled-hover);--mantine-primary-color-light: var(--mantine-color-red-light);--mantine-primary-color-light-hover: var(--mantine-color-red-light-hover);--mantine-primary-color-light-color: var(--mantine-color-red-light-color);--mantine-primary-color-0: var(--mantine-color-red-0);--mantine-primary-color-1: var(--mantine-color-red-1);--mantine-primary-color-2: var(--mantine-color-red-2);--mantine-primary-color-3: var(--mantine-color-red-3);--mantine-primary-color-4: var(--mantine-color-red-4);--mantine-primary-color-5: var(--mantine-color-red-5);--mantine-primary-color-6: var(--mantine-color-red-6);--mantine-primary-color-7: var(--mantine-color-red-7);--mantine-primary-color-8: var(--mantine-color-red-8);--mantine-primary-color-9: var(--mantine-color-red-9);}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-red-4);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-red-6);}
    </style>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
    </style>
    <div>
      Mocked RouteProvider
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
